app/scripts/modules/amazon/src/loadBalancer/configure/common/LoadBalancerLocation.tsx

```diff
@@ -111,51 +115,102 @@ class LoadBalancerLocationImpl extends React.Component<ILoadBalancerLocationProp
   }
 
   public componentDidMount(): void {
-    if (this.props.isNew || this.props.forPipelineConfig) {
-      this.loadAccounts();
-      const hideInternalFlag = this.shouldHideInternalFlag();
-      this.setState({ hideInternalFlag });
-    }
-
+    this.setState({ hideInternalFlag: this.shouldHideInternalFlag() });
     if (this.props.loadBalancer && this.props.isNew) {
       this.buildName();
     }
-  }
 
-  private loadAccounts(): void {
-    AccountService.listAccounts('aws').then(accounts => {
-      this.setState({ accounts });
-      this.accountUpdated(this.props.formik.values.credentials);
+    const formValues$ = this.props$.map(props => props.formik.values);
+    const selectedAccount$ = formValues$.map(x => x.credentials).distinctUntilChanged();
+    const selectedRegion$ = formValues$.map(x => x.region).distinctUntilChanged();
+    const subnetPurpose$ = formValues$.map(x => x.subnetType).distinctUntilChanged();
+
+    const appName$ = this.props$.map(props => props.app.name).distinctUntilChanged();
```

Comment:
Seems like this would never change and maybe doesn't need to be a stream

Reply:
Yeah, it probably doesn't need to be a stream in this context, but I think of this as defensive coding.  If this component were ever to be used in a different context where the app prop could change, this code would ensure the component still works as expected.

Reply:
👍